scipy/special/_basic.py

```diff
@@ -2943,7 +2943,8 @@ def corr(k, r): return np.power(k, -r / k) / gamma(r / k + 1) * r
     for r in np.unique(n_mod_k):
         if r == 0:
             continue
-        result[n_mod_k == r] *= corr(k, r)
+        # cast to int because uint types break on `-r`
+        result[n_mod_k == r] *= corr(k, int(r))
```

Comment:
Thanks for the quick fix @h-vetinari, I was just wondering about the performance of individually casting the entries of `n_mod_k` vs setting the dtype of  `n_mod_k` earlier on only if it is an unsigned dtype?

Reply:
We're only cycling over `np.unique(n_mod_k)`, which has at most `k` entries. I don't think it's relevant TBH. 🤷 

Also, casting `n_mod_k` is quite a bit harder, because there's no good choice of dtype to cast to - aside from not knowing the size, there's also no signed integer type that contains all the values of `uint64`. Casting with `int()` avoids all these questions.

Reply:
Fine as long as possibility has been considered :+1: